src/wfgoto.cpp

```diff
@@ -95,61 +96,55 @@ bool CompareNodes(const PDNODE& a, const PDNODE& b)
 	return ParentOrdering(a, b) < 0;
 }
 
-vector<PDNODE> *FilterBySubtree(vector<PDNODE> * parents, vector<PDNODE> * children)
+vector<PDNODE> FilterBySubtree(vector<PDNODE> const& parents, vector<PDNODE>  const& children)
 {
-	if (parents == NULL || children == NULL)
-		return NULL;
-
-	vector<PDNODE> *results = new vector <PDNODE>();
+	vector<PDNODE> results;
 
 	// for each child, if parent in parents, return
-	for (int i = 0; i < children->size(); i++)
+	std::copy_if(children.begin(),
+				 children.end(),
+				 std::back_inserter(results),
+				 [&parents](PDNODE const& child)

```

Comment:
If we're in C++14 mode why not use `const auto&`? Does having the type here provide value?